## Homework 4

Modern Stellar Astrophysics <br>
Melissa Ness <br>
December 11th 2018 <br>
Yasmeen Asali

In [3]:
import numpy as np
import astropy.units as u

### Problem 3

Comparing the adiabatic and radiative gradients: 

$$ \nabla_{ad}  = \frac{\gamma - 1}{\gamma}$$

$$ \nabla_{rad} = \frac{P}{T}\frac{dT}{dP} $$

Plugging in the value of gamma:

$$ \nabla_{ad}  = \frac{2}{5} $$

To find the value of $ \nabla_{rad} $, we can use the equation for opacity $ \kappa = \kappa_{o} \rho T^{-3.5} $, and plug that into the ideal gas law:

$$ P = \frac{R \kappa T^{4.5}}{\mu \kappa_{o}} $$

$$ \frac{dP}{dT} = 4.5 \frac{R \kappa T^{3.5}}{\mu \kappa_{o}} $$

$$ \frac{dP}{dT} = 4.5 \frac{P}{T} $$

$$ \frac{P}{T}\frac{dT}{dP} = \frac{1}{4.5}$$

$$ \nabla_{rad} = \frac{2}{9} $$

Since $\frac{2}{9} < \frac{2}{5}$, we see that the radiative gradient is less than the adiabatic one, which means that the star is convectively stable. 

### Problem 4

**Part a)**

The equation for outward radiation force is given by:

$$ F_{rad} = \frac{\kappa L}{4 \pi R^{2}c} $$

And gravitational force per unit mass at the surface where $r=R$ is given by:

$$ F_{grav} = \frac{G M}{R^{2}} $$

The limit where the blob is ejected from the star occurs when the outward radiation force becomes strong enough to overcome the gravitational attraction that keeps the blob attached to the surface. Mathematically, it means $ F_{rad} >  F_{grav}$. 

$$ \frac{G M}{R^{2}} < \frac{\kappa L}{4 \pi R^{2}c} $$

$$ {G M} < \frac{\kappa L}{4 \pi c} $$

$$ \frac{M}{L} < \frac{\kappa}{4 \pi c G} $$

**Part b)**

The max luminosity is 

$$ L = \frac{4 \pi c G M}{\kappa} $$

In [18]:
c = 3e10 * (u.cm / u.s)
G = 6.67 * 10**(-8) * ((u.cm)**3 / (u.g * (u.s)**2))
k = 0.3 * ((u.cm)**2 / u.g)

lum_const = (4*np.pi*c*G)/k
print(lum_const)

83817.69199777568 cm2 / s3


So the constants give a value of 83817, which we can multiply by mass to give:

$$ L_{max} = 83817 {M} \frac{cm^{2}}{s^{3}} $$

If mass is given in grams, this will give us units of $\frac{erg}{s} $:

$$ L_{max} = 83817 {M} \frac{erg}{s} $$

So to put this in terms of solar units: 

$$ L_{max} = 83817 \frac{M} {M_{\odot}} L_{\odot} \frac{erg}{s} $$

In [21]:
L_sol = 3.826 * 10**33 * (u.erg/u.s)

print(lum_const * L_sol)

3.2068648958348976e+38 cm2 erg / s4


So our final answer is:

$$ L_{max} = 3.2 * 10^{38} \frac{M} {M_{\odot}} \frac{erg}{s} $$

Note that the units are kind of weird, but $ erg = \frac{g cm^{2}}{s^{2}} $ so I think this works out somehow. Honestly, I am a little bit confused, but I think the value I obtained is correct so I think I just made a mistake with units along the way.  

This is the called the Eddington luminosity. 

** Part c) **

$$ \frac{L}{L_{\odot}} = \Big( \frac{M}{M_{\odot}} \Big)^{4} $$

So using the values from above:

$$ 83817 \frac{M} {M_{\odot}} = \Big( \frac{M}{M_{\odot}} \Big)^{4} $$

$$ 83817 = \Big( \frac{M}{M_{\odot}} \Big)^{3} $$

$$ M = \Big( 83817 \Big)^{1/3} M_{\odot} $$

In [25]:
M_limit = (lum_const)**(1/3) * M_sol

print(M_limit) #issue with units

8.704994813629282e+34 cm(2/3) g / s


### Problem 5

** Part a) **

The rate of mass loss will increase as the luminosity goes up, which intuitively makes sense since radiation can drive stellar winds that drive mass away from the star. As the rate of mass loss increases, the mass of the star gets smaller faster. Additionally, as gravity decreases the tenuous outer layers of the star will feel less attracted to the star, which means that they are more likely to be ejected. Radius is also inversely proportional. Intuitively, it would make sense that since gravity is stronger at smaller radii, mass loss would decrease as radius decreases. However since gravity is inversely propotional to the square of radius, the factor of $\frac{1}{r^{2}}$ will cancel with the factor of $r$ to give a proportionality of $\frac{1}{r}$. 

** Part b) **

First we calculate radius:

$$ L = 4 \pi R^{2} \sigma T^{4} $$

$$ R = (\frac{L}{4 \pi \sigma T^{4}})^{1/2} $$

In [11]:
L = 7000 * (3.826 * 10**33 * (u.erg/u.s))
T = 3000 * u.K
sigma = 5.6704 * 10**(-5) * (u.erg / (u.cm**2 * u.s * u.K**4))

R = np.sqrt(L / (4 * np.pi * sigma * T**4))
print(R)

21541067096026.137 cm


Next we have to calculate the surface gravity using this formula: 

$$ g = \frac{GM}{R^{2}} $$

In [13]:
M_sol = 1.9891 * 10**33 * u.g
g = (G*M_sol) / (R**2)
print(g)

0.28592229715002837 cm / s2


The last step is plugging in our values for g and R into the mass loss formula to solve for the mass loss rate of a solar mass star. Since $\eta$ is near unity, we can let it equal one and exclude it from calculations.

In [15]:
g_sol = 2.74 * 10**4 * (u.cm / u.s**2)
L_over_Lsol = 7000
g_over_gsol = g / g_sol
R_over_Rsol = R / (6.955 * 10**10 * u.cm)

mass_loss = (- 4 * 10**(-13) * (L_over_Lsol)) / ((g_over_gsol) * (R_over_Rsol))
print(mass_loss) 

-8.663442367505272e-07


So our final mass loss rate is $ \boxed{ -8.66 * 10^{-7} M_{\odot} yr^{-1} } $. 

** Part c) **

Expressing $\dot{M}$ as a function of $L, R, M$:

$$ \dot{M} = -4 * 10^{-13} \eta \frac{ \frac{L}{L_{\odot}} }{ (\frac{g}{g_{\odot}})(\frac{R}{R_{\odot}}) } M_{\odot} yr^{-1} $$

Let's start by finding an expression for $\frac{g}{g_{\odot}}$

$$ \frac{g}{g_{\odot}} = \frac{ \frac{GM}{R^{2}} }{ \frac{GM_{\odot}}{R_{\odot}^{2}} }$$

$$ \frac{g}{g_{\odot}} = \frac{M}{M_{\odot}} \frac{R_{\odot}^{2}}{R^{2}}$$

Substituting that back in:

$$ \dot{M} = -4 * 10^{-13} \eta \frac{ \frac{L}{L_{\odot}} }{ (\frac{M}{M_{\odot}} \frac{R_{\odot}^{2}}{R^{2}})(\frac{R}{R_{\odot}}) } M_{\odot} yr^{-1} $$

$$ \dot{M} = -4 * 10^{-13} \eta \frac{ \frac{L}{L_{\odot}} }{ \frac{M}{M_{\odot}} \frac{R_{\odot}}{R} } M_{\odot} yr^{-1} $$

$$ \dot{M} = -4 * 10^{-13} \eta \frac{ \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} }{ \frac{M}{M_{\odot}} } M_{\odot} yr^{-1} $$

Or in a format that's easier to read

$$\boxed{ \dot{M} = -4 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} \frac{M_{\odot}}{M} M_{\odot} yr^{-1} }$$

** Part d) **

$$\frac{dM}{dt} = -4 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} \frac{M_{\odot}}{M} M_{\odot} yr^{-1}$$

We can treat this as a seperable differential equation. Let's integrate from the starting mass $M_{o}$ to $M(t)$ for a given time $t$. I'm also gonna remove the units $yr^{-1}$ for simplicity. 

$$\int_{M_{o}}^{M(t)} \frac{M}{M_{\odot}} {dM} = -4 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} \int_{0}^{t}{dt} M_{\odot} $$

$$\frac{M(t)^{2}}{2M_{\odot}^{2}} - \frac{M_{o}^{2}}{2M_{\odot}^{2}} = -4 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} t $$

$$\frac{M(t)^{2} - M_{o}^{2}}{2M_{\odot}^{2}} = -4 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} t $$

$$ M(t)^{2} = M_{o}^{2} - 8 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} t M_{\odot}^{2} $$

$$ \boxed{ M(t) = \Big( M_{o}^{2} - 8 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} M_{\odot}^{2} t \Big)^{1/2} } $$

** Part e) **

For a $M_{o} = 1 M_{\odot}$ star to lose it's envelope, it looses $0.4 M_{\odot}$ of mass and leaves behind a $0.6 M_{\odot}$ core.

$$ 0.6 M_{\odot} = \Big( M_{\odot}^{2} - 8 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} M_{\odot}^{2} t \Big)^{1/2} $$

$$ 0.6 M_{\odot} = \Big( 1 - 8 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} t \Big)^{1/2} M_{\odot} $$

$$ 0.6 = \Big(1 - 8 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}} t \Big)^{1/2} $$

$$ \Big( 0.6 \Big)^{2} = 1 - 8 * 10^{-13} \eta \frac{L}{L_{\odot}} \frac{R}{R_{\odot}}t $$

$$ t = \frac{(0.6)^{2}}{ 1 - 8 * 10^{-13} \eta \frac{L}{L_{\odot}}\frac{R}{R_{\odot}}}$$

Solving for this value (letting $\eta = 1$):

In [23]:
time = ((0.6)**2) / (1 - (8 * 10**(-13) * L_over_Lsol * R_over_Rsol))

print(time)

0.3600006243977942


So the answer is $ \boxed{ t = 0.36 yr^{-1} }$